src/Command/Cleanup.php

```diff
@@ -54,15 +55,23 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $indices = $this->getIndices();
 
         foreach ($indices as $index) {
+            if (!$this->shouldProcessNonBundleIndex($index)) {
+                continue;
+            }
+
             $client = $this->esClient->getIndex($index);
 
+            if ($client->getSettings()->getBool('hidden')) {
+                continue;
+            }
+
             foreach ($client->getAliases() as $alias) {
                 $client->removeAlias($alias);
             }
 
             try {
                 $client->delete();
-            } catch (ResponseException $e) {
+            } catch (ElasticsearchException $e) {
                 $this->output->writeln(sprintf('<error>%s</error>', $e->getMessage()));
```

Comment:
In the past usually only `$client->delete()` needed error handling as only this call had errors that needed to be caught i.e. errors that occurred because the ES cluster/index was in an unexpected state.